src/Autosuggest.js

```diff
@@ -136,6 +136,15 @@ export default class Autosuggest extends Component {
       if (this.willRenderSuggestions(nextProps)) {
         if (this.state.isCollapsed && !this.justSelectedSuggestion) {
           this.revealSuggestions();
+        } else if (!nextProps.highlightFirstSuggestion) {
+          this.resetHighlightedSuggestion();
+        }
+        if (
+          this.state.highlightedSuggestionIndex === 0 &&
+          this.props.highlightFirstSuggestion &&
+          !nextProps.highlightFirstSuggestion
+        ) {
+          this.resetHighlightedSuggestion();
```

Comment:
It looks like you've added the discussed change here, but also left the previous code too. Is that intentional? I think the `else if` case above shouldn't be here now, unless I'm missing something.